backend/src/main/java/com/google/sps/datastore/UserDatastore.java

```diff
@@ -0,0 +1,80 @@
+package com.google.sps.datastore;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.data.User;
+import javax.annotation.Nullable;
+
+/** Separates datastore method calls involving User type from caller */
+public final class UserDatastore {
+
+  // Datastore Key/Property constants
+  private static final String KIND_USER = "User";
+  private static final String PROPERTY_USERNAME = "username";
+
+  /** Datastore */
+  private final DatastoreService datastore;
+
+  /** Constructor that takes in DatastoreService */
+  public UserDatastore(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  /** Put user in datastore. Overwrite user entity if user with same username already exists. */
+  public void addUser(User user) {
+    // Set properties of entity based on user fields
+    Entity userEntity = new Entity(KIND_USER, user.getUsername());
+    userEntity.setProperty(PROPERTY_USERNAME, user.getUsername());
+
+    // Insert entity into datastore
+    datastore.put(userEntity);
+  }
+
+  /** Return User Entity from username, or null if entity is not found */
+  @Nullable
+  private Entity getEntityFromUsername(String username) {
```

Comment:
tiny nit: 'from' suggests that the entity is extracted somehow from the String. I would recommend `getEntityForUsername` or even just `getEntity`